src/Chainweb/Chainweb/MinerResources.hs

```diff
@@ -80,14 +82,16 @@ withMiningCoordination
 withMiningCoordination logger enabled cutDb inner
     | not enabled = inner Nothing
     | otherwise = do
-        t <- newTVarIO mempty
-        c <- newIORef 0
-        fmap snd . concurrently (prune t c) $ inner . Just $ MiningCoordination
+        ms <- newTVarIO mempty
+        ec <- newIORef 0
+        cs <- newTVarIO . cycle . HS.toList $ chainIds cutDb
```

Comment:
`cycle` is supposed to be constant time / space.

Reply:
This makes me nervous.  What are we trying to accomplish here?

Reply:
This seems like a recipe for disaster of someone inadvertently forces this.  We need a more obviously foolproof solution.